### DISCUSSION TITLE
Time based release schedule

### DISCUSSION THREAD

**Discussion by bertfrees** [daisy/pipeline#684] [General]
From now on, DAISY Pipeline will have four releases a year, on prearranged dates. Previously, new versions were released on a ad hoc basis.

Reasons for the new strategy are:

- **Structure & predictability**: Users know when to expect a new release and developers have a date to work towards. Our development efforts are aimed at providing the promised features at that date[^1].
- **Better planning & visibility of upcoming features**: Before, users didn't really know when to expect specific features because the high-level operational plan is made up only once a year. By splitting up the plan into four parts, creating Github issues and assigning them to specific milestones, the upcoming features become more visible to everyone[^2]. Work items that can not be planned in advance, such as newly discovered critical bugs or new high-priority or easy to fix feature requests, do not appear in the high-level plan, but are part of the Github planning.
- **More agile and responsive development cycle**: Users who submit bug reports or feature requests do not have to wait a year to see the effect on the planning. It helps to maintain engagement from and dialog with users.

For now the aim is to release in ISO weeks 3, 14, 24/25 and 41/42, that means mid January, begin April, mid June and mid October. The precise due date of the upcoming release is announced in advance.

[^1]: It is important to note that with time-based releases it is always possible that features shift to a next release, however the intent remains to stick with the plan as much as possible, and if features need to be moved do it sufficiently in advance. 
[^2]: In addition to Github milestones, the idea is to provide the list of upcoming features also on a wiki page or pinned discussion, for those who are less familiar with Github issues.

**Comment by egli** [comment]
Very nice, thank you. 
I would remove the Next, Hot and After milestones. Afaikt these are no longer relevant and cause confusion

**Reply by bertfrees** [reply]
Good point.

**Comment by avneeshsingh** [comment]
Thanks Bert.
One clarification to add. The annual workplan of Pipeline project will continue to provide the high level direction in alignment to the strategy and collective longer term goals of DAISY Consortium.
And we will be working with the Pipeline user base through out the year for making adjustments and prioritizing as well as de-prioritizing the tasks.
